squidpy/gr/_build.py

```diff
@@ -46,43 +47,48 @@ def spatial_neighbors(
     coord_type
         Type of coordinate system. Valid options are:
 
-            - `{c.GRID!r}` - grid coordinates.
-            - `{c.GENERIC!r}` - generic coordinates.
+            - `{c.GRID.s!r}` - grid coordinates.
+            - `{c.GENERIC.s!r}` - generic coordinates.
+            - `None` - `{c.GRID.s!r}` if ``spatial_key`` is in :attr:`anndata.AnnData.uns`
+              with ``n_neighs = 6`` (Visium), otherwise use `{c.GENERIC.s!r}`.
+    n_neighs
+        Depending on the ``coord_type``:
 
-        If `None`, use `{c.GRID!r}` if ``spatial_key`` is present in :attr:`anndata.AnnData.uns`,
-        with `neigh_grid=6` (Visium), otherwise use `{c.GENERIC!r}`.
-    n_rings
-        Number of rings of neighbors for grid data. The argument is used if `coord_type={c.GRID!r}`.
-    neigh_grid
-        Number of neighboring tiles in a grid. The argument is used if `coord_type={c.GRID!r}`.
-    n_neigh
-        Number of neighborhoods to consider for non-grid data. The argument is used if `coord_type={c.GENERIC!r}`.
-    delaunay
-        Whether to compute the graph from Delaunay triangulation. The argument is used if `coord_type={c.GENERIC!r}`.
+            - `{c.GRID.s!r}` - number of neighboring tiles.
+            - `{c.GENERIC.s!r}` - number of neighborhoods for non-grid data. Only used when ``delaunay = False``.
     radius
-        Radius of neighbors for non-grid data. The argument is used if `coord_type={c.GENERIC!r}`.
+        Depending on the type:
+
+            - :class:`float` - neighborhood radius for non-grid data if ``coord_type = {c.GENERIC.s!r}``
+              and ``delaunay = False``.
```

Comment:
current behaviour does not prune the graph if `delaunay=True`. Therefore, specifying `delaunay=False` in docs is confusing. I think the current behaviour is fine (just prune when Tuple is passed) but maybe should be clarified a bit?

Proposal:
```
- :class:`float` - compute the graph based on neighborhood radius. Only used when ``coord_type = {c.GENERIC.s!r}``.

- :class:`tuple` - prune the final graph to only contain edges in interval `[radius[0], radius[1]]`. Can be applied to any ``coord_type = {c.GENERIC.s!r}`` mode of graph construction.
```

the `Only used when ``coord_type = {c.GENERIC.s!r}``.` think is useful to make it consistent with delaunay

Reply:
> current behaviour does not prune the graph if delaunay=True.

I think I get it now - I just wanted to highlight that it's impossible to only specify 1 `float` when `delaunay = True`, which for KNN is the number of neighborhood radius (to threshold it, you need both min and max, as for all other cases).
I will remove the `delaunay` mention in this case.

Re the `tuple` change you've suggested: it's not correct, you can also threshold using `grid` to build the graph and I fail to see why it shouldn't (see the case of `n_rings > 1`).

Reply:
>  it's not correct, you can also threshold using grid to build the graph and I fail to see why it shouldn't (see the case of n_rings > 1).

this I haven't tried, but if that's the case, I'd remove this behaviour. The idea of "grid" is that it completely disregards the "unit measure" and just build graph based on a regular grid (according to data). The `min_dist` between 2 nodes ina regular grid can be 0.1 or 1000 but it doesn't matter, as a grid graph will be built. Therefore, there is no meaning for a "radius" value in grid, because a radius is based on a "distance" in some "unit measure". In summary, imho there shouldn't be any method from "generic" that also work on "grid".

Reply:
happy to discuss further if you disagree btw, but to the logics are very different to me, as well as the generated manifolds

Reply:
No I think it makes sense - guess I was confused why we then output both the similart and distances in this case as well.